packages/kbn-alerts-grouping/src/alerts_grouping/components/alerts_grouping.tsx

```diff
@@ -0,0 +1,260 @@
+import type { Filter } from '@kbn/es-query';
+import { isNoneGroup, useGrouping } from '@kbn/grouping';
+import { isEqual } from 'lodash/fp';
+import { i18n } from '@kbn/i18n';
+import { useAlertDataView } from '@kbn/alerts-ui-shared';
+import { AlertsGroupingLevel } from './alerts_grouping_level';
+import { AlertsGroupingProps } from '../types';
+import {
+  AlertsGroupingContextProvider,
+  useAlertsGroupingState,
+} from '../contexts/alerts_grouping_context';
+import { useGroupingPersistence } from '../hooks/use_grouping_persistence';
+import { DEFAULT_PAGE_INDEX, DEFAULT_PAGE_SIZE, MAX_GROUPING_LEVELS } from '../constants';
+
+const AlertsGroupingInternal = (props: AlertsGroupingProps) => {
+  const {
+    groupingId,
+    services,
+    featureIds,
+    defaultGroupingOptions,
+    defaultFilters,
+    globalFilters,
+    globalQuery,
+    renderGroupPanel,
+    getGroupStats,
+    children,
+  } = props;
+  const { storage, dataViews, notifications, http } = services;
+  const { grouping, updateGrouping } = useAlertsGroupingState(groupingId);
+
+  const { dataViews: alertDataViews } = useAlertDataView({
+    featureIds,
+    dataViewsService: dataViews,
+    http,
+    toasts: notifications.toasts,
+  });
+  const dataView = useMemo(() => alertDataViews?.[0], [alertDataViews]);
+  const { getStoragePageSize, setStoragePageSize } = useGroupingPersistence(storage, groupingId);
+
+  const onOptionsChange = useCallback(
+    (options) => {
+      // useGrouping > useAlertsGroupingState options sync
+      // the available grouping options change when the user selects
+      // a new field not in the default ones
+      updateGrouping({
+        options,
+      });
+    },
+    [updateGrouping]
+  );
+
+  const { getGrouping, selectedGroups, setSelectedGroups } = useGrouping({
+    componentProps: {
+      groupPanelRenderer: renderGroupPanel,
+      getGroupStats,
+      unit: (totalCount) =>
+        i18n.translate('alertsGrouping.unit', {
+          values: { totalCount },
+          defaultMessage: `{totalCount, plural, =1 {alert} other {alerts}}`,
+        }),
+    },
+    defaultGroupingOptions,
+    fields: dataView?.fields ?? [],
+    groupingId,
+    maxGroupingLevels: MAX_GROUPING_LEVELS,
+    onOptionsChange,
+  });
+
+  useEffect(() => {
+    // The `none` grouping is managed from the internal selector state
+    if (isNoneGroup(selectedGroups)) {
+      // Set active groups from selected groups
+      updateGrouping({
+        activeGroups: selectedGroups,
+      });
+    }
+  }, [selectedGroups, updateGrouping]);
+
+  useEffect(() => {
+    if (!isNoneGroup(grouping.activeGroups)) {
+      // Set selected groups from active groups
+      setSelectedGroups(grouping.activeGroups);
+    }
+  }, [grouping.activeGroups, setSelectedGroups]);
+
+  const [pageIndex, setPageIndex] = useState<number[]>(
+    Array(MAX_GROUPING_LEVELS).fill(DEFAULT_PAGE_INDEX)
```

Comment:
I am a bit confused by this piece of code. What is the reason we need an array with the same values?